src/aiida/cmdline/commands/cmd_node.py

```diff
@@ -488,32 +488,33 @@ def graph_generate(
     show,
     output_file,
 ):
-    """Generate a graph from a ROOT_NODE (specified by pk or uuid)."""
+    """Generate a graph from ROOT_NODES (specified by pk or uuid)."""
     from aiida.tools.visualization import Graph
 
     link_types = {'all': (), 'logic': ('input_work', 'return'), 'data': ('input_calc', 'create')}[link_types]
 
     echo.echo_info(f'Initiating graphviz engine: {engine}')
     graph = Graph(engine=engine, node_id_type=identifier)
-    echo.echo_info(f'Recursing ancestors, max depth={ancestor_depth}')
-
-    graph.recurse_ancestors(
-        root_node,
-        depth=ancestor_depth,
-        link_types=link_types,
-        annotate_links='both',
-        include_process_outputs=process_out,
-        highlight_classes=highlight_classes,
-    )
-    echo.echo_info(f'Recursing descendants, max depth={descendant_depth}')
-    graph.recurse_descendants(
-        root_node,
-        depth=descendant_depth,
-        link_types=link_types,
-        annotate_links='both',
-        include_process_inputs=process_in,
-        highlight_classes=highlight_classes,
-    )
+
+    for root_node in root_nodes:
+        echo.echo_info(f'Recursing ancestors, max depth={ancestor_depth}')
+        graph.recurse_ancestors(
+            root_node,
+            depth=ancestor_depth,
+            link_types=link_types,
+            annotate_links='both',
+            include_process_outputs=process_out,
+            highlight_classes=highlight_classes,
+        )
+        echo.echo_info(f'Recursing descendants, max depth={descendant_depth}')
+        graph.recurse_descendants(
+            root_node,
+            depth=descendant_depth,
+            link_types=link_types,
+            annotate_links='both',
+            include_process_inputs=process_in,
+            highlight_classes=highlight_classes,
+        )
```

Comment:
Below, the default output file name is still relying on the `root_node` variable. This will still work since `root_node` is now defined inside the for loop and so the last node will be taken. But maybe we should make this code more explicit by either doing
```python
output_file = pathlib.Path(f'{root_nodes[0].pk}.{engine}.{output_format}')
```
So the code is not so implicitly dependent on the for-loop variable. Or, maybe better, we include all root nodes in the filename:
```
pks = '.'.join(n.pk for n in root_nodes)
output_file = pathlib.Path(f'{pks}.{engine}.{output_format}')
```
But maybe this can get us into trouble when the user specifies a lot of root nodes (or ones with pks that have a lot of digits) and the file name becomes larger than the file system allows. Not sure how likely this is though, and users could always explicitly define an output filename I guess.